src/components/HelpText/index.js

```diff
@@ -0,0 +1,127 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier, useDisclosure } from '../../libs/hooks';
+import InternalOverlay from '../InternalOverlay';
+import RenderIf from '../RenderIf';
+import { ESCAPE_KEY } from '../../libs/constants';
+import {
+    StyledTooltip,
+    StyledTitle,
+    StyledIconContainer,
+    StyledText,
+    StyledButton,
+} from './styled';
+import {
+    ErrorIcon,
+    InfoIcon,
+    QuestionIcon,
+    WarningIcon,
+    ErrorInverseIcon,
+    InfoInverseIcon,
+    QuestionInverseIcon,
+    WarningInverseIcon,
+} from './icons';
+
+const iconMap = {
+    question: <QuestionIcon />,
+    info: <InfoIcon />,
+    error: <ErrorIcon />,
+    warning: <WarningIcon />,
+};
+
+const inverseIconMap = {
+    question: <QuestionInverseIcon />,
+    info: <InfoInverseIcon />,
+    error: <ErrorInverseIcon />,
+    warning: <WarningInverseIcon />,
+};
+
+/**
+ * HelpText is a popup that displays information related to an element.
+ */
+const HelpText = props => {
+    const { id, title, text, variant, icon: iconProp, tabIndex, className, style } = props;
+
+    const triggerRef = useRef();
+    const helpTextId = useUniqueIdentifier('help-text');
+    const { isOpen, open: openOverlay, close: closeOverlay } = useDisclosure(false);
+
+    const handleKeyPressed = event => {
+        if (event.keyCode === ESCAPE_KEY) {
+            event.preventDefault();
+            closeOverlay();
+        }
+    };
+
+    const icon = iconProp || iconMap[variant] || iconMap.info;
+    const inverseIcon = inverseIconMap[variant] || inverseIconMap.info;
+
+    return (
+        <div id={id} className={className} style={style}>
+            <StyledButton
+                ref={triggerRef}
+                onMouseEnter={openOverlay}
+                onMouseLeave={closeOverlay}
+                onFocus={openOverlay}
+                onBlur={closeOverlay}
+                onKeyDown={handleKeyPressed}
+                type="button"
+                tabIndex={tabIndex}
+                ariaLabelledby={helpTextId}
+            >
+                {icon}
+            </StyledButton>
+            <InternalOverlay
+                isVisible={isOpen}
+                render={() => {
+                    return (
+                        <StyledTooltip id={helpTextId} role="tooltip">
+                            <RenderIf isTrue={!!title}>
```

Comment:
The `title` always should be a `string` type, if the title is passed we render the text with the icon and if is not passed return null.